src/core/Tree/TreeNode.tsx

```diff
@@ -0,0 +1,215 @@
+  const subNodeIds = context?.subNodeIds ?? [];
+
+  const styleDepth = React.useMemo(
+    () =>
+      getWindow()?.CSS?.supports?.(`--level: ${nodeDepth}`)
+        ? { '--level': nodeDepth, ...style }
+        : { marginLeft: nodeDepth ? nodeDepth * 28 : 0, ...style },
+    [nodeDepth, style],
+  );
+
+  const onNodeClick = () => {
+    if (isDisabled) {
+      return;
+    }
+    onNodeSelected?.(nodeId, !isSelected);
+  };
+
+  return (
+    <>
+      <li
+        role='treeitem'
+        id={nodeId}
+        aria-expanded={isExpanded}
+        aria-disabled={isDisabled}
+        {...rest}
+      >
+        {
+          <div
+            className={cx(
+              'iui-tree-node',
+              {
+                'iui-active': isSelected,
+                'iui-disabled': isDisabled,
+              },
+              className,
+            )}
+            style={styleDepth}
+            onClick={onNodeClick}
+            onDoubleClick={() => {
+              if (hasSubNodes) {
+                onNodeExpanded(nodeId, !isExpanded);
+              }
+            }}
+            tabIndex={isSelected ? 0 : -1}
+          >
+            {nodeCheckbox && React.isValidElement(nodeCheckbox)
+              ? React.cloneElement(nodeCheckbox, {
+                  className: cx(
+                    'iui-tree-node-checkbox',
+                    nodeCheckbox.props.className,
+                  ),
+                })
+              : nodeCheckbox}
+            <div className='iui-tree-node-content'>
+              {hasSubNodes && (
+                <IconButton
+                  styleType='borderless'
+                  size='small'
+                  onClick={(e) => {
+                    onNodeExpanded(nodeId, !isExpanded);
+                    e.stopPropagation();
+                  }}
+                  disabled={isDisabled}
+                >
+                  <SvgChevronRight
+                    className={cx('iui-tree-node-content-expander-icon', {
+                      'iui-tree-node-content-expander-icon-expanded': isExpanded,
+                    })}
+                  />
+                </IconButton>
+              )}
+              {icon &&
+                React.cloneElement(icon, {
+                  className: cx(
+                    'iui-tree-node-content-icon',
+                    icon.props.className,
+                  ),
+                })}
+              <span className='iui-tree-node-content-label'>
+                <div className='iui-tree-node-content-title'>{label}</div>
+                {sublabel && (
+                  <div className='iui-tree-node-content-caption'>
+                    {sublabel}
+                  </div>
+                )}
+              </span>
+              {children}
+            </div>
+          </div>
+        }
+        {hasSubNodes && (
+          <ul
+            className='iui-sub-tree'
+            role='group'
+            aria-owns={subNodeIds.join(', ')}
+          />
+        )}
```

Comment:
Are you sure that this `aria-owns` property does something useful for screen readers? The flat tree structure does not follow the accessibility guidelines, so my guess is that the addition of this group does not produce the desired effect.

Reply:
I hope it should 😀
![image](https://user-images.githubusercontent.com/36186912/153594813-54f28727-b4e2-420a-a856-b6d798f066db.png)

I also added some additional `aria-` attributes that are automatically added by the browser on semantically correct tree.